allauth/account/forms.py

```diff
@@ -223,17 +233,31 @@ def __init__(self, *args, **kwargs):
         super(BaseSignupForm, self).__init__(*args, **kwargs)
         # field order may contain additional fields from our base class,
         # so take proper care when reordering...
+        self.email_field_name = 'email'
         field_order = ['email', 'username']
+        if app_settings.SIGNUP_EMAIL_VERIFICATION:
+            del self.fields["email"]
```

Comment:
Is there a reason to swap out `email` fields?

If someone wanted to switch the SIGNUP_EMAIL_VERIFICATION from True to False or False to True, keeping the first field name consistent would result in a better user experience for anyone who had the form open during the setting swap.

(True to False: the second field would just be ignored, False to True: the form would redisplay with the email as entered, but would now also ask for the email to be reentered).